xtext/editorserver/src/websockets.js

```diff
@@ -0,0 +1,129 @@
+import { WebSocketServer } from 'ws';
+import { config } from "./config.js";
+import  fs from "fs";
+
+function checkEditor(data){
+
+            const editorId = data;
+            const filePath = config.deployFileLocation + "/" + editorId;
+            const buildLogPath= `${config.buildFileLocation}/${editorId}/build.log`;
+            const buildStatusPath= `${config.buildFileLocation}/${editorId}/build.res`;
+            const editorDeployed = fs.existsSync(filePath);
+            
+            let buildLog;
+            let buildStatus;
+
+            // Read the build log
+            try {
+                if (fs.existsSync(buildLogPath)) {
+                    buildLog = fs.readFileSync(buildLogPath, 'utf8');
+                } else {
+                    buildLog = "";
+                }
+            } catch (err) {
+                console.log("Error reading build log: " + buildLogPath);
+                console.log(err);
+            }
+
+            // Read the build status
+            try {
+                if (fs.existsSync(buildStatusPath)) {
+                    buildStatus = Number( fs.readFileSync(buildStatusPath, 'utf8') );
+                } else {
+                    buildStatus = 0;
+                }
+            } catch (err) {
+                console.log("Error reading build status: " + buildStatusPath);
+                console.log(err);
+            }
+
+            let response = {};
+            response.editorReady = editorDeployed;
+            response.output = buildLog;
+
+            if ( buildStatus > 0 ){
+                // Build failed
+                response.error = "Please refer to the build log.";
+            }
+	 return response;
+
+}
+
+
+function subscribe_to_build(editorID) {
+    var response = get_response();
+
+    const filePath = config.deployFileLocation + "/" + editorID;
+    const buildPath = config.buildFileLocation + "/" + editorID;
+    const buildStatusPath= `${config.buildFileLocation}/${editorID}/build.res`;
+    
+    var buildLog = '';
+    var buildStatus = -1;
+
+    // watch the build log file for changes
+    fs.watch(buildPath, () => {
+        const buildLogPath = buildPath + "/" + "build.log";
+
+        if (fs.existsSync(buildLogPath)) {
+            buildLog = fs.readFileSync(buildLogPath, 'utf8');
+        }
+
+        if (fs.existsSync(buildStatusPath)) {
```

Comment:
Why not use the event parameters, which will tell you exactly which files have changed?